### ISSUE TITLE
UI does not update if Divert Mode is enabled via MQTT

### ISSUE THREAD

**Issue by FCampos1977** [OpenEVSE/openevse_esp32_firmware#233]
If we use MQTT to set Eco Mode, system works well, but i'm not able to see this in WebUI. 
Sometimes when we check WebUI, we found OpenEVSE in Normal Mode, but is in Eco Mode.
Reboot of Wifi module, solve the problem, but few days after is the same.

**Comment by glynhudson** [comment]
I can confirm I'm able to replicate this bug. Enabling divert via MQTT e.g 

`<base-topic>divertmode/set` `2`

Results in the divert mode being successfully enabled, but the UI does not update

**Comment by larroy** [comment]
I'm unable to set divert_active with GUI v2. I see in mqtt that divert_active is false. I haven't figured out how to set it in the GUI, the only way I made it work is with mqtt setting divert_active = true

**Comment by larroy** [comment]
Using divertmode = 2 (Production) current is not modulated at all.

**Comment by larroy** [comment]
![Selection_001](https://github.com/OpenEVSE/ESP32_WiFi_V4.x/assets/928489/d5db049d-20dc-42b2-b134-7ec491bb24c4)![Selection_002](https://github.com/OpenEVSE/ESP32_WiFi_V4.x/assets/928489/e55d00d1-56e4-47cc-be37-0abc8b581049)

**Comment by larroy** [comment]
debug output shows:

```
{"divert_update":0,"solar":0,"charge_rate":0,"trigger_current":6,"voltage":240.7,"available_current":0,"smoothed_available_current":0,"pilot":25,"min_charge_end":0}

```

**Comment by larroy** [comment]
Tried latest build from git b8d1addc4fcf47c51b1c559e5a10d9c0f382bfff and I made it work.

**Comment by larroy** [comment]
Minimum charge time doesn't seem to be respected if smoothing and attack tau are 0

**Comment by larroy** [comment]
I think there's a bug. If mqtt_solar is set, then grid_ie is not updated even if divert_type is DIVERT_TYPE_GRID

**Comment by KipK** [comment]
Yep it was the case until recent dev builds. Previous divert config was set by matt topics so mqtt_solar needs to be empty if mqtt_grid is set.
UI was preventing doing this, but if you set using api it will be an issue. 
Now dev branch have a config flag for that and you can keep booth fields filled

**Comment by larroy** [comment]
I compiled myself the latest master. I will do some changes that I can share. The current logic changes current too often.

**Comment by jeremypoulter** [comment]
Not yet, there is however the divert_sim that can run various test cases. You will have to look at the GitHub action for build instructions. I have not had a chance to write more human friendly instructions.
